ntp.go

```diff
@@ -76,8 +76,12 @@ type ntpTime uint64
 // and returns the corresponding time.Duration value.
 func (t ntpTime) Duration() time.Duration {
 	sec := (t >> 32) * nanoPerSec
-	frac := (t & 0xffffffff) * nanoPerSec >> 32
-	return time.Duration(sec + frac)
+	frac := (t & 0xffffffff) * nanoPerSec
+	nsec := frac >> 32
+	if uint32(frac) >= 0x80000000 {
+		nsec++
+	}
```

Comment:
It's not 500ms, but 0.5ns. Sort of rounding.

Reply:
It is not compared to the fractional part, we compare only last 32 bits of frac * 1e9 and we truncate (loose) this bits by left shift (frac >> 32): https://github.com/beevik/ntp/blob/master/ntp_test.go#L184
NTP time is 15258.789 but we read it as 15258ns
It's because time.Duration has 1ns accuracy, but NTP time has 1/2^32 ~ 0.2328ns.

This fix makes nanosecond rounding, so 15258.789 read as 15259ns.

Reply:
Ah, got it. I got confused and didn't realize you had changed the meaning of the `frac` variable.  (The original was shifted; your version is unshifted.)

This makes sense now. Thanks for the explanation.